docs/maintaining.md

```diff
@@ -92,3 +92,118 @@ made to always install the latest released version.
+
+For `grafana/prometheus` maintainers to create a new feature, they will do the
+following:
+
+1. Create a feature branch in `grafana/prometheus` based on the latest release
+   tag that `grafana/prometheus` currently has a release branch for. The feature
+   branch should follow the naming convention `feat-<feature name>`.
+2. Implement the feature and open a PR to merge the feature branch into the
+   associated `grafana/prometheus` release branch.
+3. Once the release branch is updated, open a PR to update `grafana/agent` to
+   use the latest release branch SHA.
+
+### Updating an Existing Feature
+
+If a feature branch that was already merged to a release branch needs to be
+updated for any reason:
+
+1. Push directly to the feature branch or open a PR to merge changes into that
+   feature branch.
+2. Open a PR to merge the new changes from the feature branch into the
+   associated release branch.
+3. Once the release branch is updated, open a PR to update `grafana/agent` to
+   use the latest release branch SHA.
+
+### Handling New Upstream Release
+
+When a new upstream `prometheus/prometheus` release is available, we must go
+through the following process:
+
+1. Create a new `grafana/prometheus` release branch named
+   `release-X.Y.Z-grafana`.
+2. For all feature branches still not merged upstream, rebase them on top of the
+   latest release. Force push them to update the `grafana/prometheus` release
+   branch.
+3. Create one or more PRs to introduce the features into the newly created
+   release branch.
+
+Once this process is completed, the previous release branch in
+`grafana/prometheus` is considered stale and will no longer be updated.
+
+### Updating the Agent's vendor
+
+The easiest way to do this is the following:
+
+1. Edit `go.mod` and change the replace directive to the release branch name.
+2. Run `go mod vendor && go mod tidy`.
+3. Commit and open a PR.
+
+### Gotchas
+
+If the `grafana/prometheus` feature is incompatible with the upstream master
+branch, an upstream PR cannot be merged due to the merge conflicts. There are a
+few ways this can be handled at the feature author's discrection:
+
+1. Rebase the feature branch to `prometheus/prometheus` master so it can be
+   merged upstream. Doing this means that we cannot have the feature in th agent
+   until a new upstream release is available containing the feature.
+2. Wait until a new `prometheus/prometheus` release is available and rebase the
+   feature branch on top. The upstream PR will now be compatible with master,
+   but this window is small and may change at any time. This makes it slower to
+   get the feature merged upstream.
+3. Create a new feature branch based off of master and open a PR for that
+   feature branch. This adds extra maintenance burden on the feature author as
+   they now have to mirror changes across two feature branches.
+
+### Open Questions
+
+If two feature branches depend on one another, there are two suggested solutions
```

Comment:
Same as above, let's settle on one. I'm happy with the first if it feels simpler.